### ISSUE TITLE
Cannot run any valet commands after composer update (Global)

### ISSUE THREAD

**Issue by bmadigan** [laravel/valet#309]
I was updating my global composer json file and now I am getting the following error:

![image](https://cloud.githubusercontent.com/assets/13608/22531881/d8140450-e8b1-11e6-925b-a53f2122bd5e.png)

I don't have the Illuminate Support file in my global composer.json file, I assume it was brought in from something else.

**Comment by adamwathan** [comment]
Yeah likely there is a collision with some other global dependency. We plan to namespace a bunch of this stuff eventually, but regardless, there is always potential for annoying mismatches when using global composer packages this way.

I recommend using this library as a replacement for `composer global require`:

https://github.com/consolidation/cgr

It will allow every global dependency to live in it's own isolated environment, so you don't have to worry about any collisions.

**Comment by bmadigan** [comment]
Thanks Adam.

Took your advice and installed cgr. Now, when I run the $>cgr I get the same thing :-)
So obviously, this isn't a valet issue. 

I will close this. Thanks again.

**Comment by adamwathan** [comment]
Try uninstalling all of your other global composer dependencies and then reinstalling them.

Just delete `~/.composer/vendor`, backup your `~/.composer/composer.json` file for reference, then delete every package in there except `cgr`.

Then run `composer global update` to reinstall cgr, and then use cgr to reinstall all the packages from your backed up `composer.json` one by one.

**Comment by bmadigan** [comment]
Thanks Adam that worked. I did have to do a few other commands. After a few tries, this is what ended up working for me

Removed everything out of `composer.json`

`rm -rf ~/.composer/vendor`
`rm -rf ~/.composer/cache`
`rm  ~/.composer/composer.lock`
`composer global require consolidation/cgr`

Then from there, I installed PHPUnit and it worked. 

`cgr phpunit/phpunit`

If found that if I did not remove the `cache` and `.lock` file, it kept loading all my previous installs.
Hope this helps anyone else out.
